### PR TITLE
Qmaps 709: category panel

### PR DESCRIPTION
**Pull request by xem**

### CODE REVIEW

File: src/proxies/panel_manager.js [hunk 1]
```diff
@@ -15,7 +16,9 @@ PanelManager.setPoi = async function(poi, options) {
     if(panel.isPoiComplient) {
       panel.setPoi(poi, options)
     } else {
-      panel.close()
+      if(options && !options.isFromList){
+        panel.close()
```
Comment (amatissart) [inline src/proxies/panel_manager.js]: what's the expected behavior here ?  
This causes the favorite panel to stay active when clicking on a POI on the map, which is probably unexpected.

File: src/panel/category_panel.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    this.addCategoryMarkers();
+
+    document.querySelector(".service_panel").classList.remove("service_panel--active")
+  }
+
+  async open (options = {}) {
+    SearchInput.minify()
+    document.querySelector('#panels').classList.add('panels--hide-services')
+    document.querySelector('#panels').classList.add('panels--category-open')
```
Comment (amatissart) [inline src/panel/category_panel.js]: these classes seem to be unused. Aren't they ?

File: src/adapters/scene_category.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      this.removeCategoryMarkers()
+    })
+  }
+
+  addCategoryMarkers(pois) {
+    this.setOsmPoisVisibility(false)
+    pois.forEach(poi => {
+      const { id, name, className, subClassName, type, latLon } = poi
+      const marker = createIcon({ className, subClassName, type }, name, true)
+      poi.marker_id = id.replace("pj:", "marker_")
```
[the remark this answers is not in the document]
Comment (xem) [reply src/adapters/scene_category.js]: well, we can't use the character ":" in an ID targeted by a JS query selector (it's a reserved character for stuff like :hover :focus etc)

File: src/adapters/scene.js [hunk 4]
```diff
@@ -71,14 +74,22 @@ Scene.prototype.initMapBox = function () {
...
   this.mb.on('load', () => {
     this.onHashChange()
     new SceneDirection(this.mb)
+    new SceneCategory(this.mb)
+    
+    const CategoryPanel = PanelManager.getCategoryPanel() || {}
+    // trigger the POIs if a category has already been selected before map being loaded.    
+    if (CategoryPanel.active) {
+      CategoryPanel.addCategoryMarkers()
+    }
```
[the remark this answers is not in the document]
Comment (xem) [reply src/adapters/scene.js]: so, you want me to replace 
this.mb.on('load', ()=>{
with
window.execOnMapLoaded(()=>{
on line 82?

File: src/proxies/panel_manager.js (hunk 1 above)
Comment (xem) [reply src/proxies/panel_manager.js]: It's expected to close the panel unless we click on a PJ list item...
I don't see any problem with the favorite panel. do you mean the category/ies panel?
Please detail the buggy scenario, I don't see anything wrong yet

File: src/adapters/scene.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (amatissart) [reply src/adapters/scene.js]: nope. But maybe we could call `execOnMapLoaded` in category_panel.js if any interaction may happen before the map is loaded.

File: src/adapters/scene_category.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (amatissart) [reply src/adapters/scene_category.js]: why not using `document.getElementById` ?

**Pull request merged**